apps/api/v2/src/modules/apps/controllers/gcal-oauth/gcal-oauth.controller.ts

```diff
@@ -0,0 +1,158 @@
+import { AppsRepository } from "@/modules/apps/apps.repository";
+import { GetUser } from "@/modules/auth/decorators/get-user/get-user.decorator";
+import { AccessTokenGuard } from "@/modules/auth/guards/access-token/access-token.guard";
+import { CredentialsRepository } from "@/modules/credentials/credentials.repository";
+import { SelectedCalendarsRepository } from "@/modules/selected-calendars/selected-calendars.repository";
+import { TokensRepository } from "@/modules/tokens/tokens.repository";
+import {
+  BadRequestException,
+  Controller,
+  Get,
+  HttpCode,
+  HttpStatus,
+  Logger,
+  NotFoundException,
+  Query,
+  Redirect,
+  Req,
+  UnauthorizedException,
+  UseGuards,
+} from "@nestjs/common";
+import { ConfigService } from "@nestjs/config";
+import { Request } from "express";
+import { google } from "googleapis";
+
+import { SUCCESS_STATUS } from "@calcom/platform-constants";
+import { ApiRedirectResponseType, ApiResponse } from "@calcom/platform-types";
+
+@Controller({
+  path: "apps/gcal",
+  version: "2",
+})
+export class GoogleCalendarOAuthController {
+  private readonly logger = new Logger("Apps: Gcal Controller");
+
+  constructor(
+    private readonly appRepository: AppsRepository,
+    private readonly credentialRepository: CredentialsRepository,
+    private readonly tokensRepository: TokensRepository,
+    private readonly selectedCalendarsRepository: SelectedCalendarsRepository,
+    private readonly config: ConfigService
+  ) {}
+
+  @Get("/oauth/redirect")
+  @Redirect(undefined, 301)
+  @UseGuards(AccessTokenGuard)
+  async redirect(@Req() req: Request): Promise<ApiRedirectResponseType> {
+    const scopes = [
+      "https://www.googleapis.com/auth/calendar.readonly",
+      "https://www.googleapis.com/auth/calendar.events",
+    ];
+    const app = await this.appRepository.getAppBySlug("google-calendar");
+    if (!app) {
+      throw new NotFoundException();
```

Comment:
Could throw the exception directly on the repository but should be ok.

Reply:
not sure repository's should be handling HTTP exceptions